src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -28928,6 +28928,159 @@ public void Init(T t)
                 Diagnostic(ErrorCode.WRN_MemberNotNull, "}").WithArguments("field").WithLocation(18, 5));
         }
 
+        [Fact, WorkItem(48996, "https://github.com/dotnet/roslyn/issues/48996")]
+        public void DisallowNullProperty_Constructor()
+        {
+            var source = @"
+#nullable enable
+using System.Diagnostics.CodeAnalysis;
+
+public class C1
+{
+    [DisallowNull, NotNull] public string? F { get; set; }
+
+    public C1() { } // 1
```

Comment:
Please add a test with a non-autoprop. For example, `[DisallowNull, NotNull] public string? F { get => ""; set {} }`.

Reply:
Manually implemented properties are expected to never get initialization warnings.